### PR TITLE
Start jstorm runner moduel in feature branch with empty runner implementation.

### PR DESCRIPTION
**Pull request by peihe**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by peihe** [comment]
R: @kennknowles

File: runners/jstorm/pom.xml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                <id>validates-runner-tests</id>
+                <phase>integration-test</phase>
+                <goals>
+                  <goal>test</goal>
+                </goals>
+                <configuration>
+                  <groups>
+                    org.apache.beam.sdk.testing.ValidatesRunner
+                  </groups>
+                  <excludedGroups>
```
Comment (kennknowles) [inline runners/jstorm/pom.xml]: Let's start with this empty, since we don't yet know what needs to be excluded.

File: runners/jstorm/pom.xml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                  <dependenciesToScan>
+                    <dependency>org.apache.beam:beam-sdks-java-core</dependency>
+                  </dependenciesToScan>
+                  <systemPropertyVariables>
+                    <beamTestPipelineOptions>
+                      [
+                        "--runner=org.apache.beam.runners.jstorm.JStormRunner"
+                      ]
+                    </beamTestPipelineOptions>
+                    <beam.spark.test.reuseSparkContext>true</beam.spark.test.reuseSparkContext>
```
Comment (kennknowles) [inline runners/jstorm/pom.xml]: This doesn't belong

File: runners/jstorm/pom.xml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                    org.apache.beam.sdk.testing.ValidatesRunner
+                  </groups>
+                  <excludedGroups>
+                    org.apache.beam.sdk.testing.UsesStatefulParDo,
+                    org.apache.beam.sdk.testing.UsesTimersInParDo,
+                    org.apache.beam.sdk.testing.UsesSplittableParDo,
+                    org.apache.beam.sdk.testing.UsesCommittedMetrics
+                  </excludedGroups>
+                  <parallel>none</parallel>
+                  <forkCount>1</forkCount>
```
Comment (kennknowles) [inline runners/jstorm/pom.xml]: Let's drop all the customization that doesn't apply yet.

File: runners/jstorm/pom.xml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+      <artifactId>jstorm-core</artifactId>
+      <version>${jstorm.core.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    
+    <!-- Beam dependencies -->
+    <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>beam-sdks-java-core</artifactId>
+      <exclusions>
```
Comment (kennknowles) [inline runners/jstorm/pom.xml]: Comment on each exclusion, even if duplicated it makes it clear to someone reviewing just a few lines.

File: runners/jstorm/pom.xml [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-dependency-plugin</artifactId>
+          <executions>
+            <execution>
+              <goals><goal>analyze-only</goal></goals>
+              <configuration>
+                <!-- disable for now during runner development -->
+                <failOnWarning>false</failOnWarning>
```
Comment (kennknowles) [inline runners/jstorm/pom.xml]: Enable this

Comment (peihe) [inline runners/jstorm/pom.xml]: I already have code from prototype, and I would like to port them over incrementally.
If I turn it on now, I have to remove and add dependencies back and forth each PR.

I just want to keep this off for the first few PRs. After having the most common dependencies used in Java, I can turn it on.

File: runners/jstorm/pom.xml (hunk 4 above)
Comment (peihe) [inline runners/jstorm/pom.xml]: (I guess this is because bindings conflict?)

added

**Comment by peihe** [comment]
PTAL

File: runners/jstorm/pom.xml (hunk 5 above)
Comment (kennknowles) [inline runners/jstorm/pom.xml]: OK, that makes sense. I just didn't want to leave it on "until master". It is fine for the first few code drops.

**Comment by kennknowles** [comment]
OK, this looks good for a start. I'll merge it to the feature branch.

**Comment by kennknowles** [comment]
Actually, can you rebase and push? (you can merge to jstorm-runner branch too)

On a feature branch, our trick of having the committer squash and then add a "This closes ..." message doesn't work. The only way to get it automatically closed is to actually merge the exact commits.

**Comment by peihe** [comment]
Retest this please

**Comment by peihe** [comment]
merged to jstorm-runner branch

**Comment by peihe** [comment]
Thanks Kenn!

**Pull request closed without merging**